### DISCUSSION TITLE
Is it OK to use singleton in rust APIs

### DISCUSSION THREAD

**Discussion by EMUNES** [fzyzcjy/flutter_rust_bridge#1418] [Q&A] [answered]
Hello. I'm trying to wrap some complex rust logic into a singleton service. Can I use the singleton in rust API functions while making sure only one instance is initialized during the flutter app lifetime?

**Comment by fzyzcjy** [comment, marked as answer]
Sure, why not! Without more information, I guess `lazy_static` may be useful for you

**Comment by EMUNES** [comment]
Thanks. This clears my concern.

**Reply by fzyzcjy** [reply]
You are welcome!
